add description about limitation (#19068)
pingcap/docs: tidb-limitations.md

This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.

> **Note:**
>

> > TiDB offers high compatibility with the MySQL protocol and syntax, including many MySQL limitations. For example, a multicolumn index in TiDB also supports a maximum of 16 columns. For more information, see [MySQL Compatibility](/mysql-compatibility.md) and the official MySQL documentation.

Reviewer: ```suggestion
> TiDB offers high compatibility with the MySQL protocol and syntax, including many MySQL limitations. For example, For example, a single index can include a maximum of 16 columns. For more information, see [MySQL Compatibility](/mysql-compatibility.md) and the official MySQL documentation.
```

This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.

> Note: TiDB is highly compatible with the MySQL protocol, and also compatible with many limitations of MySQL, such as a maximum of 16 columns is permitted for multicolumn indexes, for more information, see [MySQL Compatibility](/mysql-compatibility.md) and offical doc of MySQL.

Reviewer 2: ```suggestion
Note: TiDB is highly compatible with the MySQL protocol and syntax, and also compatible with many limitations of MySQL, such as a maximum of 16 columns is permitted for multicolumn indexes, for more information, see [MySQL Compatibility](/mysql-compatibility.md) and official documentation of MySQL.
```
